src/kaboom.ts

```diff
@@ -5185,9 +5186,20 @@ export default (gopt: KaboomOpt = {}): KaboomCtx => {
 
 			game.root.get().forEach((obj) => {
 				if (!obj.stay) {
-					game.root.remove(obj)
+					game.root.remove(obj);
+				} else {
+					if (obj.scenesToStay != null) {
+						var inScenes = false;
+						for (var scene of obj.scenesToStay)
+							if (scene === id)
+								inScenes = true;
+						if (!inScenes)
+							game.root.remove(obj);
+					} else {
+						game.root.remove(obj);
+					}
```

Comment:
nit: I think these can be simplified to
```js
if (!obj.stay) {
	game.root.remove(obj)
} else {
	// if scenesToStay is defined, only stay on those scenes
	if (obj.scenesToStay != null && !obj.scenesToStay.contains(id)) {
		game.root.remove(obj)
	}
}
```
also can you run `npm run fmt`? Kaboom by default doesn't use semicolons and `fmt` can take care of that

Reply:
Should be fixed & simplified. Hopefully isn't too simplified, could fix if needed